src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,167 @@
+    private static final String VALID_PASSWORD = "*".repeat(VALID_PASSWORD_LENGTH);
+    private static final String EXCEPTION = InvalidDataException.class.toString();
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void registerValidUser_Ok() {
+        User newUser = new User();
+        newUser.setAge(VALID_AGE);
+        newUser.setLogin(VALID_LOGIN);
+        newUser.setPassword(VALID_PASSWORD);
+        registrationService.register(newUser);
+        User actual = storageDao.get(VALID_LOGIN);
+        Assertions.assertEquals(newUser, actual,
+                "Valid user should be added");
+    }
+
+    @Test
+    void registerUserWithEmptyLogin_NotOk() {
+        User newUser = new User();
+        newUser.setAge(VALID_AGE);
+        newUser.setPassword(VALID_PASSWORD);
+        newUser.setLogin("");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(newUser),
+                String.format("%S is expected when login is empty", EXCEPTION));
+    }
+
+    @Test
+    void registerNullUser_NotOk() {
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(null),
+                String.format("%s is expected when user is null", EXCEPTION));
+    }
+
+    @Test
+    void registerUserWithNullLogin_NotOk() {
+        User newUser = new User();
+        newUser.setAge(VALID_AGE);
+        newUser.setPassword(VALID_PASSWORD);
+        newUser.setLogin(null);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(newUser),
+                String.format("%s is expected when login is null", EXCEPTION));
+    }
+
+    @Test
+    void registerLoginAlreadyExist_NotOk() {
+        User newUser = new User();
+        newUser.setAge(VALID_AGE);
+        newUser.setLogin(VALID_LOGIN);
+        newUser.setPassword(VALID_PASSWORD);
+        registrationService.register(newUser);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(newUser),
+                String.format("%s is expected when user login already exist in storage",
+                        EXCEPTION));
+    }
+
+    @Test
+    void registerUserWithNullAge_NotOk() {
+        User newUser = new User();
+        newUser.setLogin(VALID_LOGIN);
+        newUser.setPassword(VALID_PASSWORD);
+        newUser.setAge(null);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(newUser),
+                String.format("%s is expected when user age is null", EXCEPTION));
+    }
+
+    @Test
+    void registerUserWithLowerAge_NotOk() {

```

Comment:
```suggestion
    void register_UserAgeLowerThenValid_NotOk() {
```
example of proper naming, apply it to all other tests